rust/processor/src/db/common/models/default_models/parquet_transactions.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use super::{
+    block_metadata_transactions::BlockMetadataTransaction,
+    parquet_write_set_changes::{WriteSetChangeDetail, WriteSetChangeModel},
+};
+use crate::{
+    parquet_processors::generic_parquet_processor::{HasVersion, NamedTable},
+    utils::{
+        counters::PROCESSOR_UNKNOWN_TYPE_COUNT,
+        util::{get_clean_payload, get_clean_writeset, get_payload_type, standardize_address},
+    },
+};
+use ahash::AHashMap;
+use allocative_derive::Allocative;
+use aptos_protos::transaction::v1::{
+    transaction::{TransactionType, TxnData},
+    Transaction as TransactionPB, TransactionInfo,
+};
+use field_count::FieldCount;
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+#[derive(Allocative, Clone, Debug, Deserialize, FieldCount, Serialize, ParquetRecordWriter)]
+pub struct Transaction {
+    pub txn_version: i64,
+    pub block_height: i64,
+    pub epoch: i64,
+    pub txn_type: String,
+    pub payload: Option<String>,
+    pub payload_type: Option<String>,
+    pub gas_used: u64,
+    pub success: bool,
+    pub vm_status: String,
+    pub num_events: i64,
+    pub num_write_set_changes: i64,
+    pub txn_hash: String,
+    pub state_change_hash: String,
+    pub event_root_hash: String,
+    pub state_checkpoint_hash: Option<String>,
+    pub accumulator_root_hash: String,
+    #[allocative(skip)]
+    pub block_timestamp: chrono::NaiveDateTime,
+}
+
+impl NamedTable for Transaction {
+    const TABLE_NAME: &'static str = "transactions";
+}
+
+impl HasVersion for Transaction {
+    fn version(&self) -> i64 {
+        self.txn_version
+    }
+}
+
+impl Default for Transaction {
+    fn default() -> Self {
+        Self {
+            txn_version: 0,
+            block_height: 0,
+            txn_hash: "".to_string(),
+            txn_type: "".to_string(),
+            payload: None,
+            state_change_hash: "".to_string(),
+            event_root_hash: "".to_string(),
+            state_checkpoint_hash: None,
+            gas_used: 0,
+            success: true,
+            vm_status: "".to_string(),
+            accumulator_root_hash: "".to_string(),
+            num_events: 0,
+            num_write_set_changes: 0,
+            epoch: 0,
+            payload_type: None,
+            #[allow(deprecated)]
+            block_timestamp: chrono::NaiveDateTime::from_timestamp(0, 0),
+        }
+    }
+}
+
+impl Transaction {
+    fn from_transaction_info(
+        info: &TransactionInfo,
+        txn_version: i64,
+        epoch: i64,
+        block_height: i64,
+    ) -> Self {
+        Self {
+            txn_version,
+            block_height,
```

Comment:
Not 100% sure the use case but if possible, it'd be good to just take in the owned object, it'd be more efficient. No worries if it doesn't make sense for this use case.